src/main/scala/org/globalforestwatch/summarystats/forest_change_diagnostic/ForestChangeDiagnosticDataLossApproxYearly.scala

```diff
@@ -0,0 +1,98 @@
+package org.globalforestwatch.summarystats.forest_change_diagnostic
+
+import frameless.Injection
+
+import scala.collection.immutable.SortedMap
+import io.circe.syntax._
+import cats.kernel.Semigroup
+import cats.implicits._
+
+import org.globalforestwatch.layers.ApproxYear
+
+// This class is for recording forest loss yearly, but also marking cases where the
+// year of loss is approximate vs exact.
+
+case class ForestChangeDiagnosticDataLossApproxYearly(value: SortedMap[ApproxYear, Double])
+  extends ForestChangeDiagnosticDataParser[ForestChangeDiagnosticDataLossApproxYearly] {
+
+  def merge(other: ForestChangeDiagnosticDataLossApproxYearly): ForestChangeDiagnosticDataLossApproxYearly = {
+    ForestChangeDiagnosticDataLossApproxYearly(Semigroup[SortedMap[ApproxYear, Double]].combine(value, other.value))
+  }
+
+  // This function combines years with both approximate and exact loss into a single
+  // year with the "approx" suffix, while a year with exact loss only is still put
+  // under the indicated year without the "approx" suffix.
+  def combineApprox: SortedMap[String, Double] = {
+    var out = scala.collection.mutable.SortedMap[String, Double]()
+    for ((k, v) <- this.value) {
```

Comment:
I think this might be easier to read with more descriptive variables, e.g. `k` as `year` and `v` as loss_area, maybe also `out` as something more descriptive

Reply:
Done!